File: frontend/service/spareparts/vehicle/GenerationService.ts
import {Generation, ProcessGeneration} from "types/spareparts/vehicle/generation";
import {Axios} from "core/axios";

export class GenerationService {

  private static readonly URL = '/generation';

  static async getAll(modelId: number | string): Promise<Generation[]> {
    const {data: generations} = await Axios.get<Generation[]>(`${this.URL}/${modelId}`);
    return generations;
  }

  static async add(processGeneration: ProcessGeneration): Promise<Generation> {
    const {data: generation} = await Axios.put<Generation>(this.URL, processGeneration);
    return generation;
  }

  static async update(id: number | string, processGeneration: ProcessGeneration): Promise<Generation> {
    const {data: generation} = await Axios.patch<Generation>(`${this.URL}/${id}`, processGeneration);
    return generation;
  }

  static async delete(id: number | string): Promise<Generation> {
    const {data: generation} = await Axios.delete<Generation>(`${this.URL}/${id}`);
    return generation;
  }

}

File: backend/src/main/java/by/minilooth/diploma/service/vehicle/GenerationService.java
package by.minilooth.diploma.service.vehicle;

import by.minilooth.diploma.exception.ActionIsImpossibleException;
import by.minilooth.diploma.exception.vehicle.GenerationNotFoundException;
import by.minilooth.diploma.exception.vehicle.ModelNotFoundException;
import by.minilooth.diploma.models.bean.vehicle.Generation;
import by.minilooth.diploma.models.bean.vehicle.Model;
import by.minilooth.diploma.models.spareparts.vehicle.ProcessGeneration;

import java.util.List;
import java.util.Optional;

public interface GenerationService {

    void save(Generation generation);
    Generation save(ProcessGeneration processGeneration);
    Generation update(ProcessGeneration processGeneration, Long id) throws GenerationNotFoundException;
    void delete(Generation generation);
    Generation delete(Long id) throws GenerationNotFoundException, ActionIsImpossibleException;
    Generation getById(Long id) throws GenerationNotFoundException;
    List<Generation> getAllByModelSorted(Long id) throws ModelNotFoundException;
    List<Generation> getAllByModel(Long id) throws ModelNotFoundException;
    List<Generation> getAll();
    Boolean existsByModel(Model model);
    Boolean existsById(Long id);

}
